packages/node-experimental/src/sdk/init.ts

```diff
@@ -30,7 +30,7 @@ if (NODE_VERSION.major && NODE_VERSION.major >= 16) {
  */
 export function init(options: NodeExperimentalOptions | undefined = {}): void {
   // Ensure we register our own global hub before something else does
-  // This will register the NodeExperimentalHub as the global hub
+  // This will register the OpenTelemetryHub as the global hub
   getCurrentHub();
```

Comment:
actually no, this was kind of a footgun I found while working on this - the first time `getCurrentHub` is called anywhere it will setup a global hub (the async context strategy also uses the global hub as a fallback), and since a bunch of the general init public APIs call `getCurrentHub` I had to call this here to ensure the global hub is actually our custom otel hub and not the regular one 😬

Reply:
OK, so some more testing: this sadly doesn't fix the issue, we still get a different global hub (from core). BUT I still refactored this so that `setOpenTelemetryContextAsyncContextStrategy` just calls `getCurrentHub` under the hood, so this is hidden away from the user and they don't need to manually call `getCurrentHub` which is a bit weird.

Reply:
Even more testing: This has weird side effects, so instead I actually export a new function `setupGlobalHub` to be called, to make it clearer what it does.